pkg/eval/str/str.go

```diff
@@ -8,6 +8,313 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare two strings and output an integer that will be 0 if a == b,
+// -1 if a < b, and +1 if a > b.
+//
+// ```elvish-transcript
+// ~> str:compare a a
+// ▶ 0
+// ~> str:compare a b
+// ▶ -1
+// ~> str:compare b a
+// ▶ 1
+// ```
+
+//elvdoc:fn contains
+//
+// ```elvish
+// str:contains $str $substr
+// ```
+//
+// If `$substr` is within `$str` output `$true` else `$false`.
```

Comment:
This still feels a bit too verbose to document a predicate. I find the following much easier to read:

Outputs whether `$substr` is within `$str`.

Also, "within" is a bit vague; I'd change that to:

Outputs whether `$str` contains `$substr` as a substring.

This also matches the order of the arguments.

Same for other predicates below.